### PR TITLE
siteAdmin: wait for all report data to load before rendering

### PR DESCRIPTION
**Pull request by bobheadxi**
closes https://github.com/sourcegraph/sourcegraph/issues/11806

Suspected race condition in `DynamicallyImportedMonacoSettingsEditor` has been causing alerts data to load inconsistently. This PR works around it by rendering a loader until all data is available.

Opened a separate issue re: potentially fixing the root cause: https://github.com/sourcegraph/sourcegraph/issues/11890

### CODE REVIEW

File: web/src/site-admin/SiteAdminReportBugPage.tsx [hunk 1]
```diff
@@ -117,23 +119,29 @@ export const SiteAdminReportBugPage: React.FunctionComponent<Props> = ({ isLight
...
+                    value={
+                        allConfig
+                            ? JSON.stringify(
+                                  monitoringStats
+                                      ? { ...allConfig, ...monitoringStats }
+                                      : { ...allConfig, alerts: null },
+                                  undefined,
+                                  2
+                              )
+                            : ''
```
Comment (felixfbecker) [inline web/src/site-admin/SiteAdminReportBugPage.tsx:135]: `allConfig` is always defined at this point afaict. If you inline the `isLoading` condition, TypeScript should be able to tell
```suggestion
                            JSON.stringify(
                                  monitoringStats
                                      ? { ...allConfig, ...monitoringStats }
                                      : { ...allConfig, alerts: null },
                                  undefined,
                                  2
                            )
```

File: web/src/site-admin/SiteAdminReportBugPage.tsx [hunk 2]
```diff
@@ -117,23 +119,29 @@ export const SiteAdminReportBugPage: React.FunctionComponent<Props> = ({ isLight
...
-                        : ''
-                }
-                jsonSchema={allConfigSchema}
-                canEdit={false}
-                height={800}
-                isLightTheme={isLightTheme}
-                history={history}
-                readOnly={true}
-            />
+            {isLoading ? (
```
Comment (felixfbecker) [inline web/src/site-admin/SiteAdminReportBugPage.tsx:122]:
```suggestion
            {allConfig === undefined || monitoringStats === undefined ? (
```

File: web/src/site-admin/SiteAdminReportBugPage.tsx [hunk 3]
```diff
@@ -92,6 +93,7 @@ export const SiteAdminReportBugPage: React.FunctionComponent<Props> = ({ isLight
     const monitoringDaysBack = 7
     const monitoringStats = useObservable(useMemo(() => fetchMonitoringStats(monitoringDaysBack), []))
     const allConfig = useObservable(useMemo(fetchAllConfigAndSettings, []))
+    const isLoading = allConfig === undefined || monitoringStats === undefined
```
Comment (felixfbecker) [inline web/src/site-admin/SiteAdminReportBugPage.tsx:96]:
```suggestion
```

File: web/src/site-admin/SiteAdminReportBugPage.tsx (hunk 1 above)
Comment (bobheadxi) [reply web/src/site-admin/SiteAdminReportBugPage.tsx]: It does indeed 😁 fixed in 6f92aa6db1cef25148ec4c70a38c8797715409a6

**Pull request merged**